spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/GcpContextConfiguration.java

```diff
@@ -95,7 +89,6 @@ public GcpContextAutoConfiguration(GcpProperties gcpProperties) {
 	}
 
 	@Bean
-	@ConditionalOnMissingBean
```

Comment:
Why removing this? It's actually come in useful on unit tests to provide mock beans to replace these ones.

Reply:
That's exactly why this PR should be handled first IMO. This is not auto-configuration so using `@ConditionalOnMissingBean` does not make any sense.

Reply:
Interesting, I think of this as auto-configuration, therefore I must be missing something.
It enables `GcpProperties` and uses them to configure the provided beans. Or, if no properties are provided, it resorts to coherent default values.

Why do you say it isn't auto-configuration?

Reply:
In `spring.factories` you have 

```
org.springframework.cloud.bootstrap.BootstrapConfiguration=\
org.springframework.cloud.gcp.core.GcpContextConfiguration
```

so it's explicitly not autoconfiguration (it's spring cloud boot strap configuration)

Reply:
I think you are focusing too much on the Cloud Config side of things. I'm almost tempted to remove the feature from `spring-cloud-gcp` altogether so we don't get too caught up (just kidding!!).

The main use of `GcpContext(Auto)Configuration` should _definitely_ be in `org.springframework.boot.autoconfigure.EnableAutoConfiguration`.
Using it in `org.springframework.cloud.bootstrap.BootstrapConfiguration` only makes sense when using Spring Cloud GCP Config. I think a minority of our users will be using it, though.

Reply:
It doesn't make sense in bootstrap configuration to use `@ConditionalOnMissingBean` (that was Stephane's point).